pulsar/backoff/backoff.go

```diff
@@ -38,6 +45,13 @@ type DefaultBackoff struct {
 	backoff time.Duration
 }
 
+func NewDefaultBackoff() Policy {
+	return &DefaultBackoff{}
+}
+func NewDefaultBackoffWithBackOff(backoff time.Duration) Policy {
+	return &DefaultBackoff{backoff: backoff}
```

Comment:
```suggestion
	return &DefaultBackoff{backoff: backoff / 2}
```

I think we should treat this as the initial backoff time, right? 
If it's, then we need to first divide the backoff by 2. Because each time Next is called, the backoff is doubled first.